src/EndpointStandalone.sol

```diff
@@ -56,11 +59,11 @@ abstract contract EndpointStandalone is
         EndpointStructs.EndpointInstruction memory instruction,
         bytes memory managerMessage
     ) external payable nonReentrant onlyManager {
-        _sendMessage(recipientChain, msg.value, msg.sender, instruction, managerMessage);
+        _sendMessage(token, recipientChain, msg.value, msg.sender, instruction, managerMessage);
     }
 
     function quoteDeliveryPrice(uint16 targetChain) external view override returns (uint256) {
-        return _quoteDeliveryPrice(targetChain);
+        return _quoteDeliveryPrice(token, targetChain);
```

Comment:
what about passing in the token as a function parameter instead of initializing a new state variable?

Reply:
Which is preferable?
Passing the token as a function parameter requires a 2 lookups against the manager contract for every transfer -- 1 lookup during quoting and 1 lookup when sending the message.
Adding the token as a state variable takes up additional state on the Endpoint but enables us to only lookup against the manager once at contract instantiation.

Reply:
Fyi I prefer the immutable, discussed in another thread